jbmc/unit/java_bytecode/java_replace_nondet/replace_nondet.cpp

```diff
@@ -110,26 +145,54 @@ void load_and_test_method(
   remove_virtual_functions(model_function);
 
   // Then test both situations.
-  THEN(
-    "Code should work when remove returns is called before "
-    "replace_java_nondet.")
+  THEN("Replace nondet should work when remove returns has been called.")
   {
     remove_returns(model_function, [](const irep_idt &) { return false; });
 
     replace_java_nondet(model_function);
 
-    validate_method_removal(goto_function.body.instructions);
+    validate_nondet_method_removed(goto_function.body.instructions);
+  }
+
+  THEN("Replace nondet should work when remove returns hasn't been called.")
+  {
+    replace_java_nondet(model_function);
+
+    remove_returns(model_function, [](const irep_idt &) { return false; });
+
+    validate_nondet_method_removed(goto_function.body.instructions);
+  }
+
+  object_factory_parameterst params{};
+
+  THEN(
+    "Replace and convert nondet should work when remove returns has been "
+    "called.")
+  {
+    replace_java_nondet(model_function);
+
+    convert_nondet(model_function, null_message_handler, params, ID_java);
+
+    remove_returns(model_function, [](const irep_idt &) { return false; });
```

Comment:
Shouldn't this happen first? (before replace and convert)

Reply:
You're right - this seems to be a copy-paste error :(